src/plot_api/plot_api.js

```diff
@@ -2634,20 +2635,20 @@ Plotly.addFrames = function(gd, frameList, indices) {
     for(i = frameList.length - 1; i >= 0; i--) {
         if(!Lib.isPlainObject(frameList[i])) continue;
 
-        var name = (_hash[frameList[i].name] || {}).name;
+        var name = (_frameHash[frameList[i].name] || {}).name;
         var newName = frameList[i].name;
 
-        if(name && newName && typeof newName === 'number' && _hash[name]) {
+        if(name && newName && typeof newName === 'number' && _frameHash[name] && numericNameWarningCount < numericNameWarningCountLimit) {
             numericNameWarningCount++;
 
-            Lib.warn('addFrames: overwriting frame "' + _hash[name].name +
+            Lib.warn('addFrames: overwriting frame "' + _frameHash[name].name +
                 '" with a frame whose name of type "number" also equates to "' +
                 name + '". This is valid but may potentially lead to unexpected ' +
                 'behavior since all plotly.js frame names are stored internally ' +
                 'as strings.');
 
-            if(numericNameWarningCount > 5) {
-                Lib.warn('addFrames: This API call has yielded too many warnings. ' +
+            if(numericNameWarningCount === numericNameWarningCountLimit) {
```

Comment:
Why `===` instead of `>` ?

Reply:
Explanation of 1st commit: it looked to me as though `Lib.warn` continued to be invoked past the 5 warnings, because I didn't see an actual condition that'd bypass this code path, so I added an extra condition in the `if` line. With the commit, if `numericNameWarningCount >= numericNameWarningCountLimit` then it doesn't descend in the `if`, so it won't warn there anymore. The condition for raising the "Too many warnings" is an equality check, because `numericNameWarningCount` is initially 0 but gets incremented past the condition check ie. before this check. If `numericNameWarningCount` is 5 here, then we've been through this conditional branch 5 times (with `numericNameWarningCount` values 1, 2, 3, 4, 5); we emit the "too many warning" things and won't come into this conditional branch again. Hope I got it right but maybe you're seeing an off-by-one error?

Reply:
Ok. That make sense. Can we add a test for this? Using jasmine `spyOn(Lib, 'warn')` should make this task relatively easy.